src/libraries/System.Net.Requests/tests/HttpWebRequestTest.cs

```diff
@@ -2318,33 +2318,38 @@ public async Task SendHttpPostRequest_When100ContinueSet_ReceivedByServer(bool e
         [ConditionalFact(typeof(RemoteExecutor), nameof(RemoteExecutor.IsSupported))]
         public async Task SendHttpRequest_WhenDefaultMaximumErrorResponseLengthSet_Success()
         {
-            await RemoteExecutor.Invoke(async (async) =>
+            await RemoteExecutor.Invoke(async isAsync =>
             {
                 TaskCompletionSource tcs = new TaskCompletionSource();
                 await LoopbackServer.CreateClientAndServerAsync(
-                async (uri) =>
+                async uri =>
                 {
                     HttpWebRequest request = WebRequest.CreateHttp(uri);
-                    HttpWebRequest.DefaultMaximumErrorResponseLength = 5;
-                    var exception = 
-                        await Assert.ThrowsAsync<WebException>(() => bool.Parse(async) ? request.GetResponseAsync() : Task.Run(() => request.GetResponse()));
+                    HttpWebRequest.DefaultMaximumErrorResponseLength = 1; // 1 KB
```

Comment:
Consider adding a test where this is set to int.MaxValue. I expect that would have caught the missing cast I commented on early, as without the cast int.MaxValue * 1024 == -1024.